tests/test_web_auth.py

```
@@ -0,0 +1,261 @@
+        "password": pytestconfig.getoption("--password"),
+    }
+
+
+@pytest.fixture(scope="session")
+def base_url(device_ip):
+    return f"http://{device_ip}"
+
+
+@pytest.fixture(scope="session", autouse=True)
+def ping_device(device_ip, base_url):
+    result = subprocess.run(
```

Ruff (0.15.11)

[error] `subprocess` call: check for execution of untrusted input

(S603)